### ISSUE TITLE
Regarding the setting issue in the comparative model

### ISSUE THREAD

**Issue by Yuwangzhi** [usail-hkust/t-PatchGNN#15]
Dear Author,  

I have read your article and found your work to be excellent and highly innovative. However, I have a few questions:  

In section "A.6.1. MODELS FOR MTS FORECASTING," you mentioned:  
“For MTS forecasting models, we input sequences after canonical pre-alignment and incorporate the observed time, mask information, and forecasting queries as additional features into these models.”  

However, despite performing canonical pre-alignment on the data, the sequence lengths in the time dimension within each batch are not consistent. For example, the input for the first batch might be (32, 41, 72), while the next batch is (32, 41, 120), where 72 and 120 represent the number of samples within the historical data used. Since these baseline models require fixed input and output sizes, how do you ensure that the inputs and outputs of these baseline models are consistent?  

Additionally, I wonder if it would be possible for you to share the code or relevant files to help me better understand and reproduce your work. If convenient, you can send the materials to my email: 1710034511@qq.com.  

Looking forward to your clarification!  

Best regards,  
Wangzhi Yu

**Comment by willzhang3** [comment]
We will pad these batches to maximal sequence lengths among them, ensuring the length consistency of inputs and outputs. You may refer to the code provided below for this sequence padding process.


```
### padding input & output to the unified length ###
B, L, K = observed_data.shape
if(L < self.max_input_len):
    pad_len = self.max_input_len - L
    observed_data = torch.cat([observed_data, self.zeros_pad[:B, :pad_len, :]], dim=1)
    observed_mask = torch.cat([observed_mask, self.zeros_pad[:B, :pad_len, :]], dim=1)
    observed_tp = torch.cat([observed_tp, self.zeros_pad[:B, :pad_len, 0]], dim=1)

_, Lp = tp_to_predict.shape
if(Lp < self.max_pred_len):
    pad_len = self.max_pred_len - Lp
    tp_to_predict = torch.cat([tp_to_predict, self.zeros_pad[:B, :pad_len, 0]], dim=1)
```

**Comment by Yuwangzhi** [comment]
ths!~
